src/shared/bus-unit-util.c

```diff
@@ -865,6 +865,9 @@ int bus_deserialize_and_dump_unit_file_changes(sd_bus_message *m, bool quiet, Un
         const char *type, *path, *source;
         int r;
 
+        assert(changes);
+        assert(n_changes);
```

Comment:
why is this necessary? neither unit_file_dump_changes() nor unit_file_changes_add() really insist on the changes ptr to be non-NULL. They both have shortcuts to become NOPs in this case, in case you don't care about the precise changes generated... Hence adding an assert() here appears wrong.